shared_model/cryptography/ed25519_sha3_impl/crypto_provider.hpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_CRYPTOPROVIDER_HPP
+#define IROHA_CRYPTOPROVIDER_HPP
+
+#include "cryptography/ed25519_sha3_impl/signer.hpp"
+#include "cryptography/ed25519_sha3_impl/verifier.hpp"
+#include "cryptography/seed.hpp"
+
+namespace shared_model {
+  namespace crypto {
+    /**
+     * Wrapper class for signing-related stuff.
+     */
+    class CryptoProvider {
```

Comment:
Have we reached any consensus on the discussions that we have to use static functions vs free functions? Sorry, I don't know much about the situation...

Reply:
I think there are no real difference between them because both are hard to mock. I introduced a class here to make it easier.